pkg/cmd/cli/cmd/process.go

```diff
@@ -172,19 +173,14 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 		return nil
 	}
 
-	label := kcmdutil.GetFlagString(cmd, "labels")
-	if len(label) != 0 {
-		lbl := ctl.ParseLabels(label)
-		for key, value := range lbl {
-			templateObj.ObjectLabels[key] = value
-		}
-	}
-
 	result, err := client.TemplateConfigs(namespace).Create(templateObj)
 	if err != nil {
 		return err
 	}
 
+	label := kcmdutil.GetFlagString(cmd, "labels")
```

Comment:
I think this needs to happen before we create the resources from the template ie. where the previous labeling was happening.